### PR TITLE
fix(auth): clear user data after signing petition

### PR DESCRIPTION
**Pull request by christopherlim98**
## Problem

Previously, users are able to still sign a petition when they refresh the page or click on the browser back button after navigating away. 

Closes [#295]

## Solution

On closing the signature modal or navigating to another page, user data will be cleared.

The screen recording shows two scenarios:
1. On user refresh
2. On user navigate away

https://user-images.githubusercontent.com/41635847/154938995-24f0d8c8-d131-4123-9224-16dfc370b758.mov

**Update 1 Mar: Added new flow and designs for pre-signing**

https://user-images.githubusercontent.com/41635847/156139919-ead22497-d27f-4881-a121-c02221eef287.mov

### CODE REVIEW

File: client/src/components/Spinner/Spinner.component.tsx [hunk 1]
```diff
@@ -5,13 +5,13 @@ import {
 } from '@chakra-ui/react'
 
 interface SpinnerProps extends ChakraSpinnerProps {
-  centerWidth?: string | number
-  centerHeight?: string | number
+  centerwidth?: string | number
+  centerheight?: string | number
```
Comment (LoneRifle) [inline client/src/components/Spinner/Spinner.component.tsx:9]: What's going on here?

**Review by LoneRifle — APPROVED** [review]
lgtm otherwise

File: client/src/components/Spinner/Spinner.component.tsx (hunk 1 above)
Comment (christopherlim98) [reply client/src/components/Spinner/Spinner.component.tsx:9]: Screenshot 2022-03-07 at 9 35 13 AM

@LoneRifle When I was debugging, I saw the react-dev-tools suggested that we use lowercase names for props. Should I revert?

Comment (LoneRifle) [reply client/src/components/Spinner/Spinner.component.tsx:9]: nah, this is okay.

**Pull request merged**